Fix kernel_restartable test for -p > 2

smharper opened the pull request:
If you run tests using -p with a number greater than two, you'll get a 'FAILED (NO EXPECTED ERR)' on restart/kernel_restartable.threads_error.  The test is expecting an error message about number of threads, but that message is never written because of an error about number of processors.  Removing the `max_parallel` limit gives the correct failure.  (I can't simply remove the `max_parallel` line because `RunException` tests default to `max_parallel = 1`)

I also fixed the `run_test -h` output

smharper committed:
Fix kernel_restartable test for -p > 2 #5430

moosebuild commented:
Results of testing 9ac014f53984c8ad2cdae205bc2ce766679f5c5e using moose_PR_pre_check recipe:

Failed on: linux-gnu

View the results here: https://www.moosebuild.org/view_job/20018

moosebuild commented:
Results of testing 9642932a8d352cb879ffc4c3d87670324cb4fb27 using moose_PR_pre_check recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.org/view_job/20023

moosebuild commented:
Results of testing 9642932a8d352cb879ffc4c3d87670324cb4fb27 using moose_PR_app_tests recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.org/view_job/20026

moosebuild commented:
Results of testing 9642932a8d352cb879ffc4c3d87670324cb4fb27 using moose_PR_test recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.org/view_job/20024

moosebuild commented:
Results of testing 9642932a8d352cb879ffc4c3d87670324cb4fb27 using moose_PR_test_dbg recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.org/view_job/20025

moosebuild commented:
Results of testing 9642932a8d352cb879ffc4c3d87670324cb4fb27 using moose_PR_modules_dbg recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.org/view_job/20027

permcody commented on python/TestHarness/TestHarness.py:
```diff
@@ -694,7 +694,7 @@ def parseCLArgs(self, argv=sys.argv[1:]):
     parser.add_argument('--dry-run', action='store_true', dest='dry_run', help="Pass --dry-run to print commands to run, but don't actually run them")
 
     outputgroup = parser.add_argument_group('Output Options', 'These options control the output of the test harness. The sep-files options write output to files named test_name.TEST_RESULT.txt. All file output will overwrite old files')
-    outputgroup.add_argument('-v', '--verbose', action='store_true', dest='verbose', help='show the output of every test that fails')
```
Not true: "-v" shows every test including passing tests.

smharper replied:
Right... so the new help text is correct

permcody commented:
Wait! Are you sure about the whole premise of this fix? In general the error messages in parallel are just not reliable if multiple processors are outputting the same message. The problem is that the error stream gets interleaved in random ways so the regex fails. This test might pass sometimes with higher numbers of processors but I don't see how it can reliably get the right message.

moosebuild commented:
Results of testing 024e1a60c94745ebd88e710c66890297d56e31f3 using moose_PR_pre_check recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.org/view_job/20042

smharper commented:
Yeah I reckon I did that kind of backwards.  I've removed the `max_parallel` from the "threads_error" test and I've added `max_parallel = 1` to "with_threads".

moosebuild commented:
Results of testing 024e1a60c94745ebd88e710c66890297d56e31f3 using moose_PR_test recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.org/view_job/20043

moosebuild commented:
Results of testing 024e1a60c94745ebd88e710c66890297d56e31f3 using moose_PR_test_dbg recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.org/view_job/20044

moosebuild commented:
Results of testing 024e1a60c94745ebd88e710c66890297d56e31f3 using moose_PR_app_tests recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.org/view_job/20045

moosebuild commented:
Results of testing 024e1a60c94745ebd88e710c66890297d56e31f3 using moose_PR_modules_dbg recipe:

Passed on: linux-gnu

View the results here: https://www.moosebuild.org/view_job/20046
